reaktive/src/commonMain/kotlin/com/badoo/reaktive/disposable/SerialDisposable.kt

```diff
@@ -0,0 +1,45 @@
+package com.badoo.reaktive.disposable
+
+import com.badoo.reaktive.utils.atomic.AtomicReference
+import com.badoo.reaktive.utils.atomic.update
+
+/**
+ * Disposable whose underlying disposable can be replaced, causing this underlying resource's disposal.
+ *
+ * When [SerialDisposable] is disposed, it's underlying disposable is disposed and all disposables assigned
+ * after that are disposed immediately.
+ */
+class SerialDisposable : Disposable {
+    private val current = AtomicReference<Disposable?>(Disposable())
```

Comment:
I noticed that `CompositeDisposable` is implemented with `synchronized` on JVM, unlike on native, please let me know if I should do the same here.

Reply:
Native `CompositeDisposable` is synchronized using atomics, same as `DisposableWrapper`.